### PR TITLE
Switch Authentication to a Token based approach

### PR DESCRIPTION
**Pull request by mabe13**
Changes the Authentication from using an API Key solution to a Token based solution.

### CODE REVIEW

**Comment by ntoll** [comment]
This looks like it'll simplify things a lot more. I guess removing the comments and adding some tests come next.

As I mentioned, there are several ways to extend the `django.contrib.auth.models.User` model, and they're rather nicely summarised here: https://simpleisbetterthancomplex.com/tutorial/2016/07/22/how-to-extend-django-user-model.html

**Comment by mabe13** [comment]
> This looks like it'll simplify things a lot more. I guess removing the comments and adding some tests come next.
> 
> As I mentioned, there are several ways to extend the `django.contrib.auth.models.User` model, and they're rather nicely summarised here: https://simpleisbetterthancomplex.com/tutorial/2016/07/22/how-to-extend-django-user-model.html

yeh tidying anything that isn't needed, removing comment code, and testing come next.

**Comment by ntoll** [comment]
:wave: I've taken a look through the code. Nice use of mocking in the unit tests. :+1:

Happily, there doesn't seem to be anything in here which jumps out at me as being particularly problematic or strange. Nice job..!

File: application/api/settings.py [hunk 1]
```diff
@@ -47,7 +47,7 @@
 
 INSTALLED_APPS = [
     "rest_framework",
-    "rest_framework_api_key",
+    "rest_framework.authtoken",
```
Comment (jope1) [inline application/api/settings.py]: Are there any pip requirement changes?

**Pull request merged**